src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,61 @@
+package duke;
+
+import duke.command.Parser;
+import duke.exception.DukeException;
+import duke.io.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Duke class encapsulates a Duke program. It contains the
+ * initialization of the Duke bot, as well as the methods
+ * required to parse and return user input.
+ */
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private TaskList taskList;
+
+    /**
+     * Constructor for a Duke bot. It also initializes the
+     * UI, Storage and tries to load an existing Task List.
+     */
+    public Duke() {
+        this.ui = new Ui();
+        this.storage = new Storage("/data/duke.txt");
+
+        try {
+            /* try loading text file from storage */
+            this.taskList = new TaskList(storage.load());
+        } catch (DukeException de) {
```

Comment:
I am not sure it is convention or not but in general all exceptions caught as written as "e".